docs/js/Expression.js

```diff
@@ -0,0 +1,59 @@
+import { UnknownOperator } from "./Errors"
+import { InputReader } from "./InputReader"
+import { Operators } from "./Operators"
+import { Brackets } from "./Brackets"
+
+
+export class Expression {
+    constructor(input) {
+        this.#reader = new InputReader(input)
+
+        this.#operands = []
+        this.#operators = []
+    }
+
+    #reader
+    #operands
+    #operators
+
+    solve() {
+        for (const token of this.#reader.iterateTokens()) {
+            if (typeof token === "number") {
+                this.#operands.push(token)
+            } else if (Brackets.isOpen(token)) {
+                this.#operators.push(token)
+            } else if (Brackets.isClosed(token)) {
+                while (!Brackets.isOpen(this.#peekOperators())) {
+                    this.#operateTop()
+                }
+                this.#operators.pop()
+            } else if (Operators.isOperator(token)) {
+                while ((this.#operators.length !== 0) &&
+                (Operators.compare(this.#peekOperators(), token) >= 0)) {
+                    this.#operateTop()
+                }
+                this.#operators.push(token)
+            } else {
+                throw new UnknownOperator(token)
```

Comment:
This `else` branch is redundant, since `Operators.apply` already throws an `UnknownOperator` Error as the default case. You can delete it and change the last `else if` to an `else`.